vms/shift/views.py

```diff
@@ -67,9 +67,10 @@ def form_valid(self, form):
         end_time = form.cleaned_data['end_time']
         shift_start_time = shift.start_time
         shift_end_time = shift.end_time
+        shift_date = shift.date
         try:
             if (end_time > start_time):

```

Comment:
I think this entire code here is badly written (not your fault, but legacy stuff). Lets clean this thing up a bit. 

1) return early. 
```
if not end_time > start_time: 
      #  the_return_commands_here

# Now, do the stuff here 
```
and now also add below it: 
``` 
if not shift_date > date.today(): 
    # the_return_command_here 
```